quinn/src/lib.rs

```diff
@@ -546,6 +323,266 @@ impl EndpointInner {
             x.notify();
         }
     }
+
+    fn poll_recv(
+        &mut self,
+        now: Instant,
+        wrapped: Rc<RefCell<EndpointInner>>,
+    ) -> Result<bool, io::Error> {
+        let mut buf = [0; 64 * 1024];
+        let mut keep_going = false;
+        let mut recvd = 0;
+        loop {
+            match self.socket.poll_recv(&mut buf) {
+                Ok(Async::Ready((n, addr, ecn))) => {
+                    if let Some(ch) = self.inner.handle(now, addr, ecn, (&buf[0..n]).into()) {
+                        self.pending.insert(ch, Pending::new(None));
+                        match self.incoming.poll_ready() {
+                            Ok(Async::Ready(())) => {
+                                self.incoming
+                                    .start_send(NewConnection::new(wrapped.clone(), ch))
+                                    .unwrap();
+                                self.inner.accept();
+                            }
+                            _ => {
+                                self.buffered_incoming.push_back(ch);
+                            }
+                        }
+                    }
+                }
+                Ok(Async::NotReady) => {
+                    break;
+                }
+                // Ignore ECONNRESET as it's undefined in QUIC and may be injected by an
+                // attacker
+                Err(ref e) if e.kind() == io::ErrorKind::ConnectionReset => {
+                    continue;
+                }
+                Err(e) => {
+                    return Err(e);
+                }
+            }
+            recvd += 1;
+            if recvd >= IO_LOOP_BOUND {
+                keep_going = true;
+                break;
```

Comment:
If we just did `return Ok(true);` here, we wouldn't need the `keep_going` variable anymore.

Reply:
Yup.